File: server/Dataprovider/Models/Book.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Dataprovider.Models;

public class Book
{
    public int BookId { get; set; }
    public string Title { get; set; } = string.Empty;

    public int AuthorId { get; set; }
    public Author Author { get; set; }
}

File: client/src/book/book.ts
export type Book = {
  bookId: number;
  title: string;
  authorId: number;
  author?: Author;
};

export type Author = {
  authorId: number;
  name: string;
};

export const defaultBook: Book = {
  bookId: 0,
  title: "",
  authorId: 1,
};
